pkg/contexts/ocm/transfer/transfer.go

```diff
@@ -125,11 +129,13 @@ func CopyVersion(printer common.Printer, log logging.Logger, hist common.History
 	log.Info("  transferring resources")
 	for i, r := range src.GetResources() {
 		var m ocm.AccessMethod
+		loop := finalize.Nested()
+
 		a, err := r.Access()
 		if err == nil {
 			m, err = r.AccessMethod()
 			if err == nil {
-				defer m.Close()
```

Comment:
Another easier option to fix this would be to simply extract this part of the function into another function in which you can defer close the AccessMethod once done.

Reply:
You are right, in this case it could be solved easier, because there is only one place where close must be called.

But the example should just show, how it can be solved  (especially for more complex ones), including the correct forwarding of error codes. just a `defer x.Close()` is basically wrong wherever it is used, because it omits the return code, nevertheless everyone uses this pattern. Go does not really has a good formal way to handle this general problem, (like the loop/defer problem).

By the way, checking the the error of the close uncovered another problem in the code, Colose is called twice.